### ISSUE TITLE
Inconsistent builds when importing elastic

### ISSUE THREAD

**Issue by knspriggs** [olivere/elastic#125]
I've had very inconsistent behaviour after starting to use the golang elasticsearch library. The first errors that cropped up were with the following error message:
`es_client.go:7:2: case-insensitive file name collision: "alias.go" and "alias.go"`

I then removed all elastic source files and tried to `go get` them again and it "hangs" on that process for as long as I can stand to wait :P. When running with `-v` I get the following output:
```
Fetching https://gopkg.in/olivere/elastic.v2?go-get=1
Parsing meta tags from https://gopkg.in/olivere/elastic.v2?go-get=1 (status code 200)
get "gopkg.in/olivere/elastic.v2": found meta tag main.metaImport{Prefix:"gopkg.in/olivere/elastic.v2", VCS:"git", RepoRoot:"https://gopkg.in/olivere/elastic.v2"} at https://gopkg.in/olivere/elastic.v2?go-get=1
gopkg.in/olivere/elastic.v2 (download)
...
gopkg.in/olivere/elastic.v2/uritemplates
gopkg.in/olivere/elastic.v2
```

Anyways, not sure what could be the cause of this, I also get similar behaviour with a forked version of the github repo as it still gets the source from gopkg.in.

Any help would be greatly appreciated.

**Comment by knspriggs** [comment]
Note I have experienced both issues with golang 1.4.2 and 1.5.

**Comment by olivere** [comment]
Hi! That's odd. I've never had this behavior. To me it sounds more of an issue with gopkg.in, but their [status page](http://stats.pingdom.com/r29i3cfl66c0) is all green. I keep this issue open and see if anyone else has further insights.

BTW: As a last resort you can manually git clone into $GOPATH/src/gopkg.in/olivere/elastic.v2 and switch to release-branch.v2. I know, I know... not for reproducible builds... just as a last resort when developing. Now with Go 1.5, you could also start [playing with vendoring](https://golang.org/s/go15vendor).

**Comment by olivere** [comment]
Can you check your git version? Gustavo Niemeyer, the author of gopkg.in, [mentioned this in the mgo-users group](https://groups.google.com/d/msg/mgo-users/rO1-gUDFo_g/DnoNIh2ADzUJ):

> And then it hangs there. No connectivity issues with gopkg.in or Github.
> ...
> This is a known but unrelated issue, and happens when using an ancient release of git (< 2.7.3.4). 
> It was fixed in this commit, specifically: 
> https://github.com/git/git/commit/d3334d9c444a22847e7add0f60c2e4a7243c151e

**Comment by knspriggs** [comment]
Thanks for the quick reply, in both of my environments I am running git 1.9.3 and 2.3.0. I suspect it is also a problem with gopkg.in, for the meantime I will clone the repo locally and work off the release-branch.v2, but check gopkg.in often to see if the issue is resolved.
Cheers!

**Comment by olivere** [comment]
Maybe it's worthwhile to post an issue on the gopkg bug tracking system and let us know here?!

**Comment by knspriggs** [comment]
Hmm, I think I've found the root issue, and its with my environment, not the source code or gopkg.in. I have a feeling that my different versions of git inside dev environment and my dev docker container (with source code volume mounted) has caused the issue. I ran some builds that added my source code to the container and then gathered dependencies inside the container and it seems to be working. Will report back, and close the issue if the problem doesn't resurface.
